src/metabase/query_processor/middleware/annotate.clj

```diff
@@ -93,16 +93,28 @@
    rf
    [(base-type-inferer metadata)]
    (fn combine [result base-types]
-     (let [cols' (mapv (fn [col base-type]
-                         (assoc col
-                                :base_type      base-type
-                                :effective_type base-type
-                                :field_ref      [:field (:name col) {:base-type base-type}]))
-                       (:cols metadata)
-                       base-types)]
-       (rf (cond-> result
-             (map? result)
-             (assoc-in [:data :cols] cols')))))))
+     (rf (if (map? result)
+           (let [name->base-type (->> (map (fn [col base-type]
+                                             (when-some [name (:lib/deduplicated-name col)]
+                                               [name base-type]))
+                                           (:cols metadata)
+                                           base-types)
+                                      (remove nil?)
```

Comment:
I don't think you need `(remove nil?)` here, `nil` as a map pair is already ignored anyway by `(into {} ...)`

```clj
(into {} [[:a 1] [:b 2] nil])
;; => {:a 1, :b 2}
```

Reply:
Obvious in the retrospect, but I've never thought of that before, thanks!